recipes/util-linux-libuuid/all/conanfile.py

```diff
@@ -109,7 +109,7 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.set_property("pkg_config_name", "uuid")
-        self.cpp_info.set_property("cmake_target_name", "LibUUID::LibUUID")
-        self.cpp_info.set_property("cmake_file_name", "LibUUID")
+        self.cpp_info.set_property("cmake_target_name", "libuuid::libuuid")
+        self.cpp_info.set_property("cmake_file_name", "libuuid")
```

Comment:
```suggestion
```

Where does it come from? Why not just keep `recipename::recipename`, it's conan-center convention when a library doesn't have any official config file or Find module file.

Reply:
This recipe provides the `libuuid` interface and replaces/deprecates the `libuuid` recipe.  The failing libraries for this migration have already been patched for `libuuid::libuuid`, so this is the smallest change that would get everything on CCI functional.

Reply:
That's not because it's the smallest change that it the good one. You can also use set_property() method of CMakeDeps in these recipes to override these names in order to satisfy these patches.

Reply:
Example if in downstream library there is a custom `Findlibuuid.cmake` and its CMakeLists.txt uses `find_package(libuuid)` and `libuuid::libuuid` target:

```python
    def generate(self):
        deps = CMakeDeps(self)
        deps.set_property("util-linux-libuuid", "cmake_find_mode", "both")
        deps.set_property("util-linux-libuuid", "cmake_file_name", "libuuid")
        deps.set_property("util-linux-libuuid", "cmake_target_name", "libuuid::libuuid")
        deps.generate()
```

Reply:
I don't disagree. I put this up to push the issue forward but I'm not particularly invested in an individual solution. I'm not sure your suggestion for `recipename::recipename` is a good one here - we're talking about the target for an interface for which there is no standard. Ideally these three libuuid recipes would be completely substitutable (independently of how advisable that would be). `recipename::recipename` doesn't satisfy that. If anything along those lines, I think it should be `provides::provides`. Whether we change what's provided is another question I guess.

Reply:
For clarity, the three libraries that I see implementing this interface are:

* https://git.kernel.org/pub/scm/fs/ext2/e2fsprogs.git/tree/lib/uuid (not in conan)
* https://github.com/util-linux/util-linux (this library)
* https://sourceforge.net/projects/libuuid/ (libuuid in conan)

See https://github.com/conan-io/conan-center-index/pull/17427#issuecomment-1545476420 for discussion of each. The original implementation is `uuid`

Reply:
> Example if in downstream library there is a custom `Findlibuuid.cmake` and its CMakeLists.txt uses `find_package(libuuid)` and `libuuid::libuuid` target:
> 
> ```python
>     def generate(self):
>         deps = CMakeDeps(self)
>         deps.set_property("util-linux-libuuid", "cmake_find_mode", "both")
>         deps.set_property("util-linux-libuuid", "cmake_file_name", "libuuid")
>         deps.set_property("util-linux-libuuid", "cmake_target_name", "libuuid::libuuid")
>         deps.generate()
> ```

I like this way of patching/modifying the conan defaults to what individual recipes require - i think it would be good to preference this over patching the build system files where possible. Having said that, given that we're talking about an implementation interface, as above, I think that the `provides` interface should be used instead of `recipename`, i.e.

```diff
-        deps.set_property("util-linux-libuuid", "cmake_find_mode", "both")
+        deps.set_property("uuid", "cmake_find_mode", "both") # or libuuid/LibUUID, whatever value `provides` has
```

Reply:
I'm coming around to using `uuid` as the `provides` value. It's the name of the lib that's built by both recipes:

https://github.com/conan-io/conan-center-index/blob/60fa77f44eafacf99b7f1d43650f13dd8db5d971/recipes/util-linux-libuuid/all/conanfile.py#L114

 https://github.com/conan-io/conan-center-index/blob/60fa77f44eafacf99b7f1d43650f13dd8db5d971/recipes/libuuid/all/conanfile.py#L88

so perhaps the target name should be `uuid::uuid`